src/testing/oem-debuggers.md

```diff
@@ -1,371 +1,493 @@
 ---
-title: Using an OEM debugger
+title: Use a native debugger
 short-title: debuggers
-description: How to connect an OEM debugger to your running Flutter app.
+description: How to connect a native debugger to your running Flutter app.
 ---
 
-If you are exclusively writing Flutter apps with
-Dart code and not using platform-specific libraries,
-or otherwise accessing platform-specific features,
-you can debug your code using your IDE's debugger.
-Only the first section of this guide, Debugging Dart code,
-is relevant for you.
-
-If you're writing a platform-specific plugin or
-using platform-specific libraries written in Swift,
-ObjectiveC, Java, or Kotlin, you can debug
-that portion of your code using Xcode (for iOS)
-or Android Gradle (for Android).
-This guide shows you how you can connect _two_
-debuggers to your Dart app, one for Dart,
-and one for the OEM code.
+<?code-excerpt path-base="testing/native_debugging"?>
 
 {{site.alert.note}}
-  If you are looking for a way to use GDB to remotely debug the
-  Flutter engine running within an Android app process,
-  check out [`flutter_gdb`][].
+  This guide presumes you understand general debugging,
+  have installed Flutter and git, and have familiarity
+  with the Dart language as well as one of the following
+  languages: Java, Kotlin, Swift, or ObjectiveC.
 {{site.alert.end}}
 
-[`flutter_gdb`]: https://github.com/flutter/engine/blob/main/sky/tools/flutter_gdb
+If you write Flutter apps only with Dart code,
+you can debug your code using your IDE's debugger.
+The Flutter team recommends VS Code.
 
-## Debugging Dart code
+If you write a platform-specific plugin or
+using platform-specific libraries, you can debug
+that portion of your code with a native debugger.
 
-Use your IDE for standard Dart debugging.
-These instructions describe Android Studio,
-but you can use your preferred IDE with the
-Flutter and Dart plugins installed and configured.
+- To debug code written in Swift or ObjectiveC, you can use Xcode.
+- To debug code written in Java or Kotlin, you can use Android Gradle.
```

Comment:
Gradle is just the build system.

Reply:
Should I say Android Studio here then?

Reply:
SGTM. I don't know what the actual debugger is called, if it has a name, but saying Android Studio is a good parallel with referring to Xcode just above.